dist/configure

```diff
@@ -296,13 +296,31 @@ if detect_arm; then
   fi
 fi
 
-if detect_x86; then
-  echo "... detected x86 platform"
+if detect_x64; then
+  echo "... detected x64 platform"
+  echo "CFLAGS_128 = -mavx" >> Makefile.config
```

Comment:
I see two issues with this approach.

This will probably work in most cases but AVX is also supported on some 32-bit systems and I'm not sure if x64 is a guarantee for avx (it's probably true in almost all cases though). But for an estimate this might be good enough.

Setting AVX on all compilation units is always a little dangerous because the compiler can add AVX instructions in parts where we don't expect it to do so. Ideally it would only be set on the files that need it. But that's a little difficult with the current setup.

Reply:
I think CFLAGS_128 is appended to CFLAGS only for the files that strictly need it, see e.g. https://github.com/project-everest/hacl-star/blob/master/dist/Makefile.tmpl#L84

we have had the issue you describe it the past :( hence this trick about CFLAGS_128 appended to CFLAGS -- note that CFLAGS_128 is platform-specific and now depends on whether we're running with AVX, NEON, or soon the vector instructions for Z and Power

as to whether we can support x86 with AVX, that's another story, indeed... and probably another PR. that seems like a corner case to me but I also don't have a good grasp of these things so it could be more important than I think